src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/OpportunitiesOrchestrator.cs

```diff
@@ -23,11 +59,73 @@ public async Task<IndexViewModel> GetIndexViewModel()
             var opportunities = opportunitiesDto.Select(x => new Opportunity
                 {
                     EmployerName = x.DasAccountName,
-                    ReferenceNumber = _encodingService.Encode(x.Id, EncodingType.PledgeId)
+                    ReferenceNumber = _encodingService.Encode(x.Id, EncodingType.PledgeId),
                 })
                 .ToList();
 
             return new IndexViewModel { Opportunities = opportunities };
         }
+
+        public async Task<string> GetUserEncodedAccountId(string userId)
+        {
+            var accounts = await _userService.GetUserAccounts(userId);
+
+            // TODO: Below is temporary -
+            //       Raised as an issue, and eventually to be replaced with
+            //       an accounts selection screen.
+            var firstEncodedAccountId = accounts
+                .Select(x => x.EncodedAccountId)
+                .First();
+
+            return firstEncodedAccountId;
+        }
+
+        public async Task<OpportunitySummaryViewModel> GetOpportunitySummaryViewModel(OpportunityDto opportunityDto, string encodedPledgeId)
+        {
+            // Pull back the tags, and use the descriptions to build the lists.
+            var sectorReferenceDataItems = await _tagService.GetSectors();
+            string sectorList = opportunityDto.Sectors.ToReferenceDataDescriptionList(sectorReferenceDataItems);
+
+            var jobRoleReferenceDataItems = await _tagService.GetJobRoles();
+            string jobRoleList = opportunityDto.JobRoles.ToReferenceDataDescriptionList(jobRoleReferenceDataItems);
+
+            var levelReferenceDataItems = await _tagService.GetLevels();
+
+            bool allContainLevel = levelReferenceDataItems.All(x => x.Id.Contains("Level"));
```

Comment:
You don't need to do anything particularly special for levels now. There's a new property on the reference data item called `ShortDescription` which will give you the levels as just numbers. If you add another extension method (or parameterise the existing one) to use this property instead, it will look much like the others.

Reply:
Fair enough - refactored what's there to make it much more simple.